app/uk/gov/hmrc/helptosavefrontend/controllers/ReminderController.scala

```diff
@@ -0,0 +1,115 @@
+                                                                                          implicit val reminderFrequencyValidation: ReminderFrequencyValidation,
+                                                                                          val frontendAppConfig:                    FrontendAppConfig,
+                                                                                          val config:                               Configuration,
+                                                                                          val env:                                  Environment,
+                                                                                          ec:                                       ExecutionContext)
+
+  extends BaseController(cpd, mcc, errorHandler) with HelpToSaveAuth with SessionBehaviour with Logging {
+
+  private def backLink: String = routes.AccessAccountController.accessAccount().url
+
+  def getSelectRendersPage(): Action[AnyContent] = authorisedForHtsWithNINO { implicit request ⇒ implicit htsContext ⇒
+
+    Ok(reminderFrequencySet(ReminderForm.giveRemindersDetailsForm(), Some(backLink)))
+
+  }(loginContinueURL = routes.ReminderController.selectRemindersSubmit().url)
+
+  def selectRemindersSubmit(): Action[AnyContent] = authorisedForHtsWithInfo { implicit request ⇒ implicit htsContext ⇒
+
+    ReminderForm.giveRemindersDetailsForm().bindFromRequest().fold(
+      withErrors ⇒ {
+        Ok(reminderFrequencySet(withErrors))
+      },
+
+      success ⇒
+        {
+
+          //Get the user name
+          val userName = htsContext.userDetails match {
+            case Left(x)  ⇒ ""
+            case Right(x) ⇒ x.forename
+
+          }
+
+          val htsUserModified = for {
+
+            emailRetrieved ← helpToSaveService.getConfirmedEmail
+
+            htsUserModified ← helpToSaveReminderService.updateHtsUser(HtsUser(Nino(htsContext.nino), emailRetrieved.getOrElse("noEmail"), userName, daysToReceive = DateToDaysMapper.d2dMapper.getOrElse(success.reminderFrequency, Seq(1))))
+
+          } yield htsUserModified
+
+          htsUserModified.fold(
+            { e ⇒
+              logger.warn(s"Could not find confirmed email: $e")
+              SeeOther(routes.EmailController.confirmEmailErrorTryLater().url)
+            },
+            htsUserModified ⇒ Ok(reminderConfirmation(htsUserModified.email, success.reminderFrequency))
+          )
+
+        }
+    )
+  }(loginContinueURL = routes.ReminderController.selectRemindersSubmit().url)
+
+  def getRendersConfirmPage(email: String, period: String): Action[AnyContent] = authorisedForHtsWithNINO { implicit request ⇒ implicit htsContext ⇒
+
+    crypto.decrypt(email) match {
+      case Success(value) ⇒ Ok(reminderConfirmation(value, period))
+      case Failure(e)     ⇒ Ok(reminderConfirmation("emailNotDecrypt", period))
```

Comment:
I'm not happy with this defaulting in the failure case, it should at least log and fail as in the uk.gov.hmrc.helptosavefrontend.controllers.EmailController#emailConfirmed